src/main/java/com/android/volley/CacheDispatcher.java

```diff
@@ -154,4 +172,104 @@ public void run() {
             }
         }
     }
+
+    private class WaitingRequestHandler implements Request.NetworkRequestCompleteListener {
```

Comment:
`private static class` - to make the interactions with CacheDispatcher explicit.

Reply:
Thanks for all the great comments! I'm new to Java, so I'm learning a lot here